GitUI/Avatars/ICachedAvatarProvider.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace GitUI.Avatars
+{
+    /// <summary>
+    /// An <see cref="IAvatarProvider"/> with support for clearing avatar caches.
+    /// </summary>
+    public interface ICachedAvatarProvider : IAvatarProvider
```

Comment:
The interface name implies one thing, but its definition provide a different behaviour. That is "cached provider" would likely do more than just _clear_ cache. Should we call this something different? Like (a random idea born at 11pm) `IAvatarCacheSweeper`?

Reply:
The way it's currently implemented is does more than just clear the cache. The new `ICachedAvatarProvider` is, from an interface standpoint, just a rename from the old `IAvatarProvider` (because it derives from the new `IAvatarProvider`)

The new `ICachedAvatarProvider` does (as the previous version of `IAvatarProvider`) two things:
- It provides avatar images
- it allows to control/clear the inner cache

So based on the current interface it is basically a "normal" AvatarProvider but with the added functionality that a potentially existing inner cache can be cleared. Renaming it to `IAvatarCacheSweeper` (or something similar) would suggest that it can only be used to clear the cache, but that's not the case (at least not at the moment).

Reply:
Sorry I was looking at it late at night, and my brain totally locked up. Of course we'd be inheriting all from `IAvatarProvider`.